### PR TITLE
Update tools, use github workflows, eslint and stylelint

### PR DESCRIPTION
**Pull request by nickvergessen**

### CODE REVIEW

File: src/public.js [hunk 1]
```diff
@@ -23,29 +23,30 @@
...
+const hasToken = (document.getElementById('sharingToken') !== null)
+const isPasswordProtected = (document.getElementById('password-submit') !== null)
 
 if (hasToken && !isPasswordProtected) {
-	$('body').prepend($('<div>').attr('id', 'terms_of_service_confirm'));
+	const tofc = document.createElement('div')
+	tofc.id = 'terms_of_service_confirm'
+	document.body.prepend(tofc)
+
 	new Vue({
```
Comment (nickvergessen) [inline src/public.js:45]: I need help here @skjnldsv 

This throws:
```
/home/nickv/Nextcloud/19/server/appsshipped/terms_of_service/src/public.js
  45:2  error  Do not use 'new' for side effects  no-new
```

but prepending `export default` like in all the other places didn't work:
```
/home/nickv/Nextcloud/19/server/appsshipped/terms_of_service/src/public.js
  45:2  error  Parsing error: 'import' and 'export' may only appear at the top level

  43 | 	document.body.prepend(tofc)
  44 | 
> 45 | 	export default new Vue({
     | 	^
  46 | 		el: '#terms_of_service_confirm',
  47 | 		data: {
  48 | 			source: 'public',
```

Comment (gary-kim) [reply src/public.js:45]: Probably better to just use `// eslint-disable-next-line`.

Comment (skjnldsv) [reply src/public.js:45]: It's because it's nested into a if
export/import needs to be at the root level of the app :)

Just disable the `no new` rule :)

Comment (nickvergessen) [reply src/public.js:46]: Thanks!

**Pull request merged**